### PR TITLE
PCHR-1452: Skipping API GET call on Contact if there is an empty array of Contact IDs.

### PR DESCRIPTION
**Pull request by coldrunKacper**
The issue appears on CiviCRM 4.7 (regarding to API changes) and causes an error when using
'id' => array('IN' => array())
API parameters with empty array (while CiviCRM 4.5 works fine with it).
I've added an additional check and now there is no call executed if ID array is empty.
Also I've removed '$rowContacts' variable initialization as it wasn't used in the script.

### CODE REVIEW

File: civihr_employee_portal/civihr_employee_portal.module [hunk 1]
```diff
@@ -6273,3 +6273,71 @@ function _is_hrreports_data_table_view($viewName) {
...
+ * 4 - DAY_AFTER_WEEKEND
+ * 5 - TOMORROW
+ * 3 - ANY_OTHER_DAY
+ * Used in Tasks block template.
+ * 
+ * @param string $date
+ * @return int
+ */
+function _get_task_filter_by_date($date) {
+    $today = date('Y-m-d');
```
Comment (nishant-bhorodia) [inline civihr_employee_portal/civihr_employee_portal.module]: Please can we use `DateTime` objects for all dates in this function for consistency or is there a specific reason why both `date` and `DateTime` are used ? Thank you!

File: civihr_employee_portal/civihr_employee_portal.module [hunk 2]
```diff
@@ -6273,3 +6273,71 @@ function _is_hrreports_data_table_view($viewName) {
...
+    $tomorrow = new DateTime('tomorrow');
+    $nbDay = date('N', strtotime($today));
+    $sunday = new DateTime($today);
+    $sunday->modify('+' . (7 - $nbDay) . ' days');
+    $weekEnd = $sunday->format('Y-m-d');
+    $taskDate = date('Y-m-d', strtotime(strip_tags($date)));
+
+    if ($taskDate < $today) {
+        return 1;
```
Comment (nishant-bhorodia) [inline civihr_employee_portal/civihr_employee_portal.module]: Perhaps use of constants will make it more readable.

**Comment by omarabuhussein** [comment]
generally looks okay .. but i have one question and i know it's not directly related to your ticket .. If I understand correctly , It appear that the contact names are pre-fetched to be used in javascript for search .. but this doesn't scale very well and ajax calls should be used instead.

**Comment by coldrunKacper** [comment]
It's difficult to say what is better practice as in this case we only want to search within currently shown records so applying ajax calls would cause returning the same set of data which we already have in the template.

**Comment by omarabuhussein** [comment]
> we only want to search within currently shown records

if the number of shown records is small then i guess it's fine ... approved

**Pull request merged**